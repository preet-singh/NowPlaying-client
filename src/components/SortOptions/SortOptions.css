
.sortOptions a:visited {
  color: white !important;
}