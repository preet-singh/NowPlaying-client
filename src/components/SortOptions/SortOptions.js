//Dependencies
import React from 'react';
import {Link} from 'react-router-dom';
import './SortOptions.css'

//Components
import FilterBox from '../FilterBox/FilterBox';

//Utilities
import UserContext from '../../utils/context';

class SortOptions extends React.Component {
  static contextType = UserContext;
  constructor(props) {
    super(props);
    this.state = {
      displayFilter: false
    }
  }

  render() {
    return(
      <div className="sortOptions">
        <div id="filterButton">
          <button type="button" className="black-button margin10" onClick={() => this.setState({displayFilter: true})}>Filter</button>
          <button type="button" className="black-button margin10" onClick={() => this.context.setSearchedCategoryItems(this.context.categoryItems)}>Clear</button>
        </div>
        {this.state.displayFilter ? <FilterBox /> : ''}
        <p className="black-bg"><Link to="/new">Movie not here? <span className="block bold">Find your movie!</span></Link></p>
      </div>
    );
  }
}

export default SortOptions;