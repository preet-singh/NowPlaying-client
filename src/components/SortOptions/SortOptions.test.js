import React from 'react';
import SortOptions from './SortOptions.js';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`SortOptions component`, () => {

  it('renders the RegisterForm', () => {
    const wrapper = shallow(<SortOptions />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})