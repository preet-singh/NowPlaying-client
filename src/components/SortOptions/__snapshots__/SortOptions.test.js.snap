// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortOptions component renders the RegisterForm 1`] = `
<div
  className="sortOptions"
>
  <div
    id="filterButton"
  >
    <button
      className="black-button margin10 filter-button"
      onClick={[Function]}
      type="button"
    >
      Filter
    </button>
    <button
      className="black-button margin10 filter-button"
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </div>
  <p
    className="black-bg"
  >
    <Link
      to="/new"
    >
      Movie not here? 
      <span
        className="block bold"
      >
        Find your movie!
      </span>
    </Link>
  </p>
</div>
`;
